client/src/components/ExportCsv.tsx

```diff
@@ -0,0 +1,85 @@
+import {formatIsoDate} from 'date/utils';
+import {ExportToCsv} from 'export-to-csv';
+
+const sum = (acc: number, item: number) => acc + item;
+
+function csvExporter(users: any) {
+  let data = users.map((row: any) => {
+    return {
+      Username: row.displayName,
+      'Change Requests': row.repoTotals
+        .map((repo: any) => repo.authoredTotals.changesCreated)
+        .reduce(sum),
+      'Comments Written (All)': row.repoTotals
+        .map((repo: any) => repo.authoredTotals.commentsWrittenTotal)
+        .reduce(sum),
+      'Comments to Others': row.repoTotals
+        .map((repo: any) => repo.authoredTotals.commentsWrittenToOthers)
+        .reduce(sum),
+      'Approvals Given': row.repoTotals
+        .map((repo: any) => repo.authoredTotals.approvals)
+        .reduce(sum),
+      'Rejections Given': row.repoTotals
+        .map((repo: any) => repo.authoredTotals.rejections)
+        .reduce(sum),
+      'Comments Received': row.repoTotals
+        .map((repo: any) => repo.receivedTotals.commentsByOthers)
+        .reduce(sum),
+      'Approvals Received': row.repoTotals
+        .map((repo: any) => repo.receivedTotals.approvals)
+        .reduce(sum),
+      'Rejections Received': row.repoTotals
+        .map((repo: any) => repo.receivedTotals.rejections)
+        .reduce(sum),
+      'Review requests': row.aggregatedReceivedTotals.reviewRequests,
+      'Comments / Change': Number.isNaN(
+        row.aggregatedReceivedTotals.commentsByOthers / row.aggregatedAuthoredTotals.changesCreated
+      )
+        ? '-'
+        : (
+            row.aggregatedReceivedTotals.commentsByOthers /
+            row.aggregatedAuthoredTotals.changesCreated
+          ).toFixed(3),
+      'Comments / Request': Number.isNaN(
+        row.aggregatedAuthoredTotals.commentsWrittenToOthers /
+          row.aggregatedReceivedTotals.reviewRequests
+      )
+        ? '-'
+        : (
+            row.aggregatedAuthoredTotals.commentsWrittenToOthers /
+            row.aggregatedReceivedTotals.reviewRequests
+          ).toFixed(3),
+      'First Seen': formatIsoDate(row.interval?.startDate),
+      'Last Seen': formatIsoDate(row.interval?.endDate),
+      'Active Days': row.activeDaysCount,
+      'Avg. time in review': Math.round(
+        row.aggregatedAuthoredTotals.meanChangeOpenTimeMsec / 1000 / 60 / 60
+      ),
+    };
+  });
+  const options = {
+    fieldSeparator: ',',
+    quoteStrings: '"',
+    decimalSeparator: '.',
+    showLabels: true,
+    showTitle: true,
+    filename: `codereview-stats.csv`,
+    useTextFile: false,
+    useBom: true,
+    useKeysAsHeaders: true,
+  };
+  const csvExporter = new ExportToCsv(options);
+  csvExporter.generateCsv(data);
+}
+
+const ExportCsv = ({users}: any) => {
+  return (
+    <div className="text-right">
```

Comment:
In this project, all top-level components have a camel-cased class name that LooksLikeThis, and matches the name of the file, so "ExportCsv" in this case. But it'd be better to name this for what it is - "ExportToCsvButton". That will help the reader to think about what the component looks like, as well as to what it does.

I'm not sure where `text-right` style is actually coming from. It'd be better to encapsulate whatever custom styles we might want in ExportToCsvButton.css, similar to what other components in the project do.

Reply:
Yeah, I should rename it to "ExportToCsvButtom". thanks for pointing that out. The `text-right` is a Bootstrap class which basically means `text-align: right`. I have not used any custom styling I just used built-in Bootstrap classes.

Reply:
Alright. Since this project doesn't use any bootstrap styles directly but uses .css files with a matching name to the component, I'd suggest we add that separate file and add that style there instead. A few more lines of code, but it's more consistent.

Reply:
Agree